pkg/snapstore/utils.go

```diff
@@ -211,3 +212,40 @@ func getLatestCredentialsModifiedTime(credentialFiles []string) (time.Time, erro
 	}
 	return latestModifiedTime, nil
 }
+
+// findFileWithExtensionInDir checks if there is any file in directory dir which has the file extension extension.
+func findFileWithExtensionInDir(dir, extension string) (string, error) {
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		return "", fmt.Errorf("error while listing files to fetch credentials in %v directory with error: %w", dir, err)
+	}
+	var credentialFile string
+	for _, dirEntry := range dirEntries {
+		if filepath.Ext(dirEntry.Name()) == extension {
+			credentialFile = filepath.Join(dir, dirEntry.Name())
+			break
+		}
+	}
+	// return the first file found with the extension
+	return credentialFile, nil
+}
+
+// getJSONCredentialModifiedTime returns the modification time of a JSON file if it is present in the directory dir.
+// This function is introduced only to support JSON files being present in the directory which is passed through the
+// PROVIDER_APPLICATION_CREDENTIAL environment variable. Will be removed by v0.31.0.
+func getJSONCredentialModifiedTime(dir, providerErrorString string) (time.Time, error) {
```

Comment:
I don't understand the point of taking `providerErrorString` as an argument as this function `getJSONCredentialModifiedTime` doesn't look like it is using this argument. It just using it in error string which I guess caller can do it.

Reply:
Alright, will make that change! Thanks for the suggestion.

Reply:
[81620c7](https://github.com/gardener/etcd-backup-restore/pull/730/commits/81620c7e6af279c5a0e631b8fa6dfa2140f58b3f).